kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ActionsIterator.java

```diff
@@ -157,19 +175,55 @@ private void tryEnsureNextActionsIterIsReady() {
         }
     }
 
+    /**
+     * Get an iterator of actions from a checkpoint manifest that may contain sidecar files. If the
+     * current read schema includes Add/Remove files, then inject the sidecar column into this
+     * schema to read the sidecar files from the checkpoint manifest. When the returned
+     * ColumnarBatches are processed, these sidecars will be appended to the end of the file list
+     * and read as part of a subsequent batch (avoiding reading the checkpoint manifest more than
+     * once).
+     */
+    private CloseableIterator<ColumnarBatch> getActionsIterFromSinglePartOrV2Checkpoint(
+            FileStatus file,
+            String fileName) throws IOException {
+        // If the sidecars may contain the current action, read sidecars from the checkpoint
+        // manifest (to be read later).
+        StructType modifiedReadSchema = readSchema;
+        if (schemaContainsAddOrRemoveFiles) {
+            modifiedReadSchema = LogReplay.withSidecarFileSchema(readSchema);
```

Comment:
Why is this line dependent on `schemaContainsAddOrRemoveFiles` ? Shouldn't we always read sidecar file entries from classic/v2 checkpoint?

Reply:
As I mentioned above, we don't always want to read sidecar file entries if we are looking only for non-file actions. In this case, we can avoid doing the extra work of injecting/dropping sidecar files from the schema.